src/jpeg/encoder.rs

```diff
@@ -775,14 +767,13 @@ fn build_quantization_segment(m: &mut Vec<u8>, precision: u8, identifier: u8, qt
     assert_eq!(qtable.len() % 64, 0);
```

Comment:
Shouldn't this be testing that qtable.len >= 64? Otherwise, it'd allow a qtable with len 0.

Reply:
Yeah, that seems correct.

Reply:
I think it could be even more specific and check for equality. It's only ever called from here

https://github.com/image-rs/image/blob/a890363ca2b6850bb9673ea9174d8042216efa0a/src/jpeg/encoder.rs#L532-L535

and that provides `chunks(64)`. That could then also be changed to `chunks_exact(64)` with its own assert?

Reply:
It looks like tables is always exactly 128 u8's long:

`       
        let mut tables = Vec::new();
        let scale_value = |&v: &u8| {
            let value = (u32::from(v) * scale + 50) / 100;

            clamp(value, 1, u32::from(u8::max_value())) as u8
        };
        tables.extend(STD_LUMA_QTABLE.iter().map(&scale_value));
        tables.extend(STD_CHROMA_QTABLE.iter().map(&scale_value));
`

and I don't know if there's anything that would make it violate this constraint, given that the elements of JpegEncoder are not publically accessible and table is only modified at instantiation - so instead of using chunks_exact, we could just do `tables: [u8; 128]` and take the chunks manually.

Reply:
I wouldn't rely on this here if it's not necessary. This method follows a part of the standard that describes how to encode the 8×8 quantization tables which justifies the length of 64. As an encoder we _can_ choose to only every use the same two tables, so that's fine. However, we do not want to unreasonably be more strict that necessary. It seems somewhat likely that we might want to use different tables in the future? If we introduce more invariants than the 64 byte blocks enshrined in the standard this would be unnecessarily ossified.

Reply:
That said, I wouldn't be opposed to using `Vec<[u8; 64]>` instead of `Vec<u8>`! But it might be tricky and more verbose to apply `scale_value` there.

Reply:
Not really - if you take a mutable reference and modify the values in place, it's relatively straightforwards:

```
        let mut tables = vec![STD_LUMA_QTABLE.clone(), STD_CHROMA_QTABLE.clone()];
        let scale_value = |v: &mut u8| {
            let value = (u32::from(*v) * scale + 50) / 100;

            *v = clamp(value, 1, u32::from(u8::max_value())) as u8;
        };
        tables[0].iter_mut().for_each(scale_value);
        tables[1].iter_mut().for_each(scale_value);
```

Reply:
Good idea, that should work :+1: